app/upgrades/README.md

```diff
@@ -10,4 +14,31 @@ This folder contains logic for every osmosis upgrade. (Both state migrations, an
 * v7 - Carbon State migration
 * v8 - Nitrogen State migration
 
-## TODO: Make a fork-upgrade struct and a state-migration upgrade struct
+## Upgrade types 
+
+There are two upgrade types exposed, `Upgrade` and `Fork`. 
+An `Upgrade` defines an upgrade that is to be acted upon by state migrations from the SDK `x/upgrade` module.
+A `Fork` defines a hard fork that changes some logic at a block height. 
+If the goal is to have a new binary be compatible with the old binary prior to the upgrade height,
+as is the case for all osmosis `Fork`s, then all logic changes must be height-gated or in the `BeginForkLogic` code.
+
+```go
+type Upgrade struct {
```

Comment:
These code snippets are likely to go out of sync with the actual code overtime

Reply:
Hopefully they never update O_O